src/test/metrics_flow_test.cpp

```diff
@@ -0,0 +1,605 @@
+    }
+
+    // ovms_requests_success
+    checkRequestsCounter(server.collect(), "ovms_requests_success", modelName, 1, "gRPC", "Predict", "TensorFlowServing", dynamicBatch * numberOfSuccessRequests + numberOfSuccessRequests);  // ran by demultiplexer + real request
+    checkRequestsCounter(server.collect(), "ovms_requests_success", dagName, 1, "gRPC", "Predict", "TensorFlowServing", numberOfSuccessRequests);                                             // ran by real request
+
+    checkRequestsCounter(server.collect(), "ovms_requests_fail", modelName, 1, "gRPC", "Predict", "TensorFlowServing", numberOfFailedRequests);  // ran by real request
+    checkRequestsCounter(server.collect(), "ovms_requests_fail", dagName, 1, "gRPC", "Predict", "TensorFlowServing", numberOfFailedRequests);    // ran by real request
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_request_time_us_count{interface=\"gRPC\",name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(numberOfSuccessRequests)));
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_request_time_us_count{interface=\"gRPC\",name=\""} + dagName + std::string{"\",version=\"1\"} "} + std::to_string(numberOfSuccessRequests)));
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_request_time_us_count{interface=\"REST\",name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(0)));
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_request_time_us_count{interface=\"REST\",name=\""} + dagName + std::string{"\",version=\"1\"} "} + std::to_string(0)));
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_inference_time_us_count{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(dynamicBatch * numberOfSuccessRequests + numberOfSuccessRequests)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_inference_time_us_count{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_wait_for_infer_req_time_us_count{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(dynamicBatch * numberOfSuccessRequests + numberOfSuccessRequests)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_wait_for_infer_req_time_us_count{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_streams{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(4)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_streams{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
+
+    EXPECT_THAT(server.collect(), HasSubstr(std::string{"ovms_infer_req_queue_size{name=\""} + modelName + std::string{"\",version=\"1\"} "} + std::to_string(2)));
+    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_infer_req_queue_size{name=\""} + dagName + std::string{"\",version=\"1\"} "})));
```

Comment:
```suggestion
    EXPECT_THAT(server.collect(), Not(HasSubstr(std::string{"ovms_infer_req_queue_size{name=\""} + dagName})));
```
This is stricter check I think